server/src/main/java/org/elasticsearch/search/aggregations/support/AggregationContext.java

```diff
@@ -277,12 +277,25 @@ public ProductionAggregationContext(
             Supplier<Boolean> isCancelled
         ) {
             this.context = context;
-            this.breakerService = new PreallocatedCircuitBreakerService(
-                context.bigArrays().breakerService(),
-                CircuitBreaker.REQUEST,
-                bytesToPreallocate
-            );
-            this.bigArrays = context.bigArrays().withBreakerService(breakerService).withCircuitBreaking();
+            if (bytesToPreallocate == 0) {
+                /*
+                 * Its possible if a bit strange for the aggregations to ask
+                 * to preallocate 0 bytes. Mostly this is for testing other
+                 * things, but we should honor it and just not preallocate
+                 * anything. Setting the breakerService reference to null will
+                 * cause us to skip it when we close this context.
+                 */
+                this.breakerService = null;
+                this.bigArrays = context.bigArrays().withCircuitBreaking();
+            } else {
+                this.breakerService = bytesToPreallocate == 0 ? null : new PreallocatedCircuitBreakerService(
```

Comment:
this ternary shouldn't be necessary, right?

Reply:
Correct! A leftover.

Reply:
I'll zap it.